app/src/main/java/com/zestworks/surveys/di/AppModule.kt

```diff
@@ -45,4 +48,24 @@ class AppModule(private val context: Context) {
     fun provideRepository(surveyDatabase: SurveyListDatabase, surveyApi: SurveyApi): SurveyRepository {
         return OfflineFirstRepository(surveyApi, surveyDatabase)
     }
+
+    @Provides
+    @PerApp
+    fun provideOkttpClient(): OkHttpClient {
+        val httpClient = OkHttpClient.Builder()
+        httpClient.addInterceptor {
+
+            val original = it.request()
+            val originalHttpUrl = original.url()
+
+            val url = originalHttpUrl.newBuilder()
+                    .addQueryParameter("access_token", BuildConfig.ACCESSTOKEN)
```

Comment:
but in most real applications this is a writable field (after a signin/signup event you usually receive an access token), any explanation why we get them from BuildConfig here?

Reply:
@sleepylee I've written a mock signup flow. We can either move our retrofit and okhttp instances outside the app module and initialise them after reading from DB/Network or do it this way, by hiding the gettoken behind an interface so that the sign in/token supply logic is oblivious of the networking dependencies. 

I've called sign in from the view because typical signins will involve contexts and a chrometab that returns a token.

Looks good? 

https://github.com/prithivraj/Surveys/pull/24